src/fromager/sources.py

```diff
@@ -209,13 +214,26 @@ def download_url(
         "looking for %s %s", outfile, "(exists)" if outfile.exists() else "(not there)"
     )
     if outfile.exists():
-        logger.debug(f"already have {outfile}")
-        return outfile
+        outfile_size_on_disk = outfile.stat().st_size
+        header_response = ctx.requests.head(url)
+        outfile_size_online = int(header_response.headers.get("Content-Length", 0))
+
+        # Check if size on disk matches the size of file online
+        if outfile_size_on_disk == outfile_size_online:
+            logger.debug(f"already have {outfile}")
+            return outfile
+        else:
+            # Set the range header
+            headers = {"Range": f"bytes={outfile_size_on_disk}-"}
+    else:
+        outfile_size_on_disk = 0
+        headers = {}
+
     # Open the URL first in case that fails, so we don't end up with an empty file.
     logger.debug(f"reading from {url}")
-    with requests.get(url, stream=True) as r:
+    with ctx.requests.get(url, stream=True, headers=headers) as r:
```

Comment:
I think we're going to need a loop here to handle the retry.

Let's set this aside for now and come back when we have more time to work on it together.